Pull Request

lhforsythe opened the pull request:
Hi there,

First off, thank you for developing such an awesome alternative launcher for WearOS. I find using this much more efficient than my OEM launcher. However, for my use case, I found the launching animation when opening applications through the launcher a bit slow and unnecessary. So, I decided to contribute to this project by creating a settings application that installs alongside the launcher to enable/disable the animation in addition to an option to send feedback (open a GitHub issue). I programmed it so that it should be easy to add more options to the settings for additional user control over the launcher. I also added a few changes that I believe help performance-wise, although possibly minimal.

I hope to work on a few more additions to the launcher later on, but I thought I would submit a pull request with what I have done thus far.

Commit:
New:
Created a settings application that installs alongside the launcher. Currently, this allows the user to change whether an animation is shown when opening an app through the launcher as well as adding the option for users to send feedback regarding the launcher via GitHub. More options within the settings application can be added.

I also (am pretty sure I) enabled HW acceleration for the launcher, as well as tweaked a few settings to allow better performance when scrolling and rearranging applications. I will work on this more, though.

Bugs: To allow the launcher settings application to show inside the launcher, I had to comment out a line of code prohibiting the launcher icon itself from showing within the launcher. This results in both the settings and launcher app/icons visible within the launcher.

Future work: I am interested in possibly recreating the Apple Watch launcher feature where you can click a clock icon inside the launcher to take you back to the watch face. Moreover, adding more options within settings and increasing performance are also things I would like to do in the future.

Commit:
Creation of Settings Application and General Tweaks

Created a settings application that installs alongside the launcher. Currently, this allows the user to change whether an animation is shown when opening an app through the launcher as well as adding the option for users to send feedback regarding the launcher via GitHub. More options within the settings application can be added.

I also (am pretty sure I) enabled HW acceleration for the launcher, as well as tweaked a few settings to allow better performance when scrolling and rearranging applications. I will work on this more, though.

Bugs: To allow the launcher settings application to show inside the launcher, I had to comment out a line of code prohibiting the launcher icon itself from showing within the launcher. This results in both the settings and launcher app/icons visible within the launcher.

Future work: I am interested in possibly recreating the Apple Watch launcher feature where you can click a clock icon inside the launcher to take you back to the watch face. Moreover, adding more options within settings and increasing performance are also things I would like to do in the future.

Commit:
Merge remote-tracking branch 'origin/master'

agronick commented:
Looks good. I test it shortly.

Commit:
Added "Back to Watch Face" Button/App

Implemented a clock button/app within the launcher that, when clicked, simulates a home button press, bringing the user back to the watch face. This is intended to be similar to how the Apple Watch's launcher functions. Overall, it functions as it should, but I may add a few bug fixes to it.

Commit:
Bug Fix:
Removed unneeded, 200ms, delay when opening applications through the launcher. This makes the launcher respond quicker to user input.

agronick commented:
I pushed a commit so the new activities don't get included in the main launcher. Once you touch up these other issues we can merge it. Thanks for the contribution.

Commit:
New Feature:

Implemented the ability to zoom out in launcher for easier access to applications. This is beneficial to users who have an immense amount of applications installed on their watch, which would otherwise take awhile to find using the normal view of the launcher.

TO DO: Implement haptic feedback when zooming out in launcher using the bezel.

Commit:
Bug Fix:

Fix to hide extra activities from main launcher (OEM provided), while allowing them to show within this launcher.

lhforsythe committed:
Update README.md

Added image/gif to README

lhforsythe commented:
> I pushed a commit so the new activities don't get included in the main launcher. Once you touch up these other issues we can merge it. Thanks for the contribution.

Apologies for the late reply. Great! I just finished up committing the bug fixes you mentioned to my main branch. I also had implemented a means of zooming out within the launcher (using the watch's bezel) and added a demonstration of this to the README file. If all is good, everything should be ready to merge.

Thanks for allowing me to contribute to the project. I actually do not have much experience with developing applications for WearOS, hence the few bugs, so this has been quite fun to work on and rather insightful. If I think of any other good ideas for the project, I'll be sure to contribute again. :)
